web/models/confirmable.ex

```diff
@@ -69,7 +69,7 @@ defmodule Coherence.Schema.Confirmable do
         """
         def confirm!(user) do
           IO.warn "#{inspect Config.user_schema}.confirm!/1 has been deprecated. Please use Coherence.ControllerHelpers.confirm!/1 instead."
-          changeset = Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil})
+          changeset = Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil, current_password: user.password})
```

Comment:
The current_password should not be required here. Please update the changeset logic so it only requires current_password validation if the password is being changed.